### PR TITLE
Glossary changes

### PR DESCRIPTION
**Pull request by fainashalts**
Added `compilation`, `contract interface` and `name` in order to have all resource objects represented. Fix a few typos and grammatical things.

### CODE REVIEW

File: docs/glossary.rst [hunk 1]
```diff
@@ -56,11 +62,14 @@ Glossary
       A placeholder representing a runtime or |Network|-specific value, at
       a particular byte offset or offsets. Identified by name.
 
-      Filled in by corresponding |LinkValue| by Truffle.
+      Filled in with corresponding |LinkValue| by Truffle.
 
    Link Value
       The value used to replace |LinkReference| placeholders.
 
+   Name
```
Comment (fainashalts) [inline docs/glossary.rst]: I basically plagiarized this from the part in the data model...

File: docs/glossary.rst [hunk 2]
```diff
@@ -35,13 +39,15 @@ Glossary
...
+      Refers to the collection of methods through which one may interact with a contract.
+
    Contract Instance
       A contract account with its own address, storage, and balance,
       representing a single deployment of a particular |ContractType|.
 
    Contract Type
-      Refers to a specific contract or library, able to deployed to a particular
-      |Network|
+      Refers to a specific contract or library, able to be deployed to a particular |Network|
```
Comment (cds-amal) [inline docs/glossary.rst]: Perhaps including a little more context would be helpful to the reader. For example: `Refers to a specific contract or library, able to be deployed to a particular |Network| as a |Contract Instance|`

This helps me realize that `Contract Type` is a Smart Contract class|source that has no deployed attributes.

Comment (fainashalts) [reply docs/glossary.rst]: This is actually not relevant anymore, and will be removed.

File: docs/glossary.rst (hunk 1 above)
Comment (gnidan) [reply docs/glossary.rst]: I don't think it's plagiarism to take words from a different section of the same document

**Pull request merged**